### PR TITLE
Use OCS Capabilities API

### LINKED ISSUE #15621: NextCloud/OwnCloud Append Connection Path to Default Path
When configuring a NextCloud/OwnCloud connection, users can easily get tripped by how the path is configured in these profiles.

With an empty path the `Home` feature is used, defaulting to `remote.php/dav/files/username`. When a path is set the `Home` feature isn't used, thus Cyberduck tries to mount `/Path/To/Some/Folder`, which should really be `remote.php/dav/files/username/Path/To/some/Folder`.

Proposal:
Unconditionally prepend "remote.php/dav/files/username" to all paths coming _from_ a connection.
Additionally: Make prefix configurable - as in 
```
<key>Properties</key>
<array>
    <string>connection.url.base=remote.php/dav/files/${username}</string>
</array>
```
Where Username-token is replaced in HostPreferences to host configured username.

### LINKED ISSUE #15874: Determine capabilities from API
As mentioned in #15860. Use cases to determine support for

- [x] Locking
- [x] Versioning
- [x] #15621
- [x] ~~#15619~~

### PR DESCRIPTION
**Pull request by dkocher**
Fix #15874.

- [ ] Failing test `OwncloudSessionTest`
- [x] Failing test `NextcloudSessionTest`

### CODE REVIEW

File: webdav/src/main/java/ch/cyberduck/core/dav/DAVSession.java [hunk 1]
```diff
@@ -150,19 +150,12 @@ public void login(final Proxy proxy, final LoginCallback prompt, final CancelCal
...
-                                host.getPort(),
-                                host.getPort(),
-                                Charset.forName(preferences.getProperty("http.credentials.charset"))
-                        );
-                        break;
-                    default:
-                        client.disablePreemptiveAuthentication();
-                }
+                // Enable preemptive authentication. See HttpState#setAuthenticationPreemptive
+                client.enablePreemptiveAuthentication(host.getHostname(),
```
Comment (ylangisc) [inline webdav/src/main/java/ch/cyberduck/core/dav/DAVSession.java:154]: No need anymore for the proxy the type check? Guess it is (or was) necessary to support authenticating proxies. Was added in 096877af2f4a2dcbae18a2cac134a40ac4843033.

File: webdav/src/main/java/ch/cyberduck/core/dav/DAVSession.java [hunk 2]
```diff
@@ -150,19 +150,12 @@ public void login(final Proxy proxy, final LoginCallback prompt, final CancelCal
...
-                        client.enablePreemptiveAuthentication(host.getHostname(),
-                                host.getPort(),
-                                host.getPort(),
-                                Charset.forName(preferences.getProperty("http.credentials.charset"))
-                        );
-                        break;
-                    default:
-                        client.disablePreemptiveAuthentication();
-                }
+                // Enable preemptive authentication. See HttpState#setAuthenticationPreemptive
```
Comment (ylangisc) [inline webdav/src/main/java/ch/cyberduck/core/dav/DAVSession.java:153]: Legacy comment. There is no `HttpState` class anymore.

**Comment by ylangisc** [comment]
> Fix #15874.
> 
> * [ ]  Failing test `OwncloudSessionTest`
> * [x]  Failing test `NextcloudSessionTest`

First task resolved too?

**Comment by dkocher** [comment]
> > Fix #15874.
> > 
> > * [ ]   Failing test `OwncloudSessionTest`
> > * [x]   Failing test `NextcloudSessionTest`
> 
> First task resolved too?

`https://demo.owncloud.org/` currently unreachable.

File: webdav/src/main/java/ch/cyberduck/core/dav/DAVSession.java (hunk 1 above)
Comment (dkocher) [reply webdav/src/main/java/ch/cyberduck/core/dav/DAVSession.java:154]: See [28fafaf](https://github.com/iterate-ch/cyberduck/pull/15876/commits/28fafaf12bb0df3e2db662742121cd3a081ecfeb). It reverts commit https://github.com/iterate-ch/cyberduck/commit/ff9c41cc85894d328d8725cc0af790a2604111bd. Proxy authentication uses `Proxy-Authentication` request header. I do not understand why this was made dependent on the proxy configuration.

Comment (ylangisc) [reply webdav/src/main/java/ch/cyberduck/core/dav/DAVSession.java:154]: Successfully tested against an authenticating proxy.

**Review by ylangisc — APPROVED** [review]

**Pull request merged**